Warn for TODO and FIXME comments

gfontenot opened the pull request:
If we find a TODO or FIXME comment while building, we should generate a
warning. This will serve as an incentive to remove these comments, and
will also reduce the likelihood that we lose track of them.

gfontenot committed:
Warn for TODO and FIXME comments

If we find a TODO or FIXME comment while building, we should generate a
warning. This will serve as an incentive to remove these comments, and
will also reduce the likelihood that we lose track of them.

tonyd256 commented on KICKOFF_PROJECT_NAME.xcodeproj/project.pbxproj:
```diff
@@ -354,6 +355,20 @@
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "KEYWORDS=\"TODO:|FIXME:|\\?\\?\\?:|\\!\\!\\!:\"\nFILE_EXTENSIONS=\"swift|h|m|mm|c|cpp\"\nfind -E \"${SRCROOT}\" -ipath \"${SRCROOT}/Carthage\" -prune -o -ipath \"${SRCROOT}/pods\" -prune -o \\( -regex \".*\\.($FILE_EXTENSIONS)$\" \\) -print0 | xargs -0 egrep --with-filename --line-number --only-matching \"($KEYWORDS).*\\$\" | perl -p -e \"s/($KEYWORDS)/ warning: \\$1/\"\n";
```
SUPER small nit pick ... lowercase `Carthage` to be consistent with `pods` even though case doesn't matter?

gfontenot replied:
I went the other way and capitalized Pods.

tonyd256 commented:
👍

gfontenot committed:
Consistently capitalize directory names

We're capitalizing Carthage we should capitalize Pods as well. This
doesn't _really_ matter since we're using a case insensitive search, but
hey, pedantry.

gfontenot commented:
ping @thoughtbot/ios

sharplet commented on KICKOFF_PROJECT_NAME.xcodeproj/project.pbxproj:
```diff
@@ -354,6 +355,20 @@
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "KEYWORDS=\"TODO:|FIXME:|\\?\\?\\?:|\\!\\!\\!:\"\nFILE_EXTENSIONS=\"swift|h|m|mm|c|cpp\"\nfind -E \"${SRCROOT}\" -ipath \"${SRCROOT}/Carthage\" -prune -o -ipath \"${SRCROOT}/Pods\" -prune -o \\( -regex \".*\\.($FILE_EXTENSIONS)$\" \\) -print0 | xargs -0 egrep --with-filename --line-number --only-matching \"($KEYWORDS).*\\$\" | perl -p -e \"s/($KEYWORDS)/ warning: \\$1/\"\n";
```
What do you think about extracting this into a separate bin script? It'd be a lot easier to manage that way than in Xcode's crappy editor, and we'd get real diffs.

gfontenot replied:
I think we're relying on some build variables set by Xcode here. What do you think about moving forward with this as is and then trying to refactor this out into a local script?

sharplet replied:
Works for me.

sharplet commented:
I'm 👍 on the idea.
